includes/Checker/Checks/Abstract_PHP_CodeSniffer_Check.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Checks\PHP_CodeSniffer_Check
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker\Checks;
+
+use WordPress\Plugin_Check\Checker\Check;
+use WordPress\Plugin_Check\Checker\Check_Result;
+use Exception;
+
+/**
+ * Check for running one or more PHP CodeSniffer sniffs.
+ *
+ * @since n.e.x.t
+ */
+abstract class Abstract_PHP_CodeSniffer_Check implements Check {
+
+	/**
+	 * List of allowed PHPCS arguments.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @var array
+	 */
+	protected $allowed_args = array(
+		'extensions',
+		'standard',
+		'sniffs',
+		'exclude',
+	);
+
+	/**
+	 * Returns an associative array of arguements to pass to PHPCS.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array
+	 */
+	abstract public function get_args();
+
+	/**
+	 * Creates the command arguments.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check results including the plugin context to check.
+	 *
+	 * @return array
+	 */
+	private function create_args( Check_Result $result ) {
+		// Set the default arguments for PHPCS.
+		$argv = array(
+			'',
+			$result->plugin()->path( '' ),
+			'--report=Json',
+			'--report-width=9999',
+		);
+
+		// Only accept allowed PHPCS arguments from check arguments array.
+		$check_args = array_filter(
+			$this->get_args(),
+			function( $key ) {
+				return in_array( $key, $this->allowed_args, true );
+			},
+			ARRAY_FILTER_USE_KEY
+		);
```

Comment:
It'd be more efficient to make `$this->allowed_args` an associative array where the keys are the actual allowed args and the values are just something like `true`. This way, you can simplify this code to use:
```suggestion
		$check_args = array_intersect_key( $this->get_args(), $this->allowed_args );
```